HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -2842,30 +2747,253 @@ def self.create_curve_quint_linear(model, coeff, name)
     return curve
   end
 
+  def self.convert_net_to_gross_capacity(net_cap, watts_per_cfm, cfm_per_ton, fan_ratio, mode)
+    net_cap_ton = UnitConversions.convert(net_cap, 'Btu/hr', 'ton')
+    net_cap_watts = UnitConversions.convert(net_cap, 'Btu/hr', 'w')
+    cfm = net_cap_ton * cfm_per_ton
+    fan_power = watts_per_cfm * cfm * (fan_ratio**3)
+    if mode == :clg
+      gross_cap_watts = net_cap_watts + fan_power
+    else
+      gross_cap_watts = net_cap_watts - fan_power
+    end
+    gross_cap_btu_hr = UnitConversions.convert(gross_cap_watts, 'w', 'Btu/hr')
+    return gross_cap_btu_hr
+  end
+
+  def self.convert_net_to_gross_capacity_cop(net_cap, fan_power, mode, output_cop, net_cop = nil)
+    net_cap_watts = UnitConversions.convert(net_cap, 'Btu/hr', 'w')
+    if mode == :clg
+      gross_cap_watts = net_cap_watts + fan_power
+    else
+      gross_cap_watts = net_cap_watts - fan_power
+    end
+    if output_cop
+      net_power = net_cap_watts / net_cop
+      gross_power = net_power - fan_power
+      gross_cop = gross_cap_watts / gross_power
+      return gross_cop
+    else
+      gross_cap_btu_hr = UnitConversions.convert(gross_cap_watts, 'w', 'Btu/hr')
+      return gross_cap_btu_hr
+    end
+  end
+
+  def self.setup_neep_detailed_performance_data(detailed_performance_data)
+    data_array = Array.new(2) { Array.new }
+    detailed_performance_data.each do |data_point|
+      # Only process min and max capacities at each outdoor drybulb
+      next unless ['minimum', 'maximum'].include? data_point.capacity_description
```

Comment:
Please make these HPXML constants (and check the code for other places they are used).